tests/test_download_cli_integration.py

```
@@ -922,24 +1093,576 @@ def test_run_download_orchestrator_none_after_init(mocker):
+    result = integration._get_tracked_desktop_versions()
+
+    assert result == {"desktop": None, "desktop_prerelease": None}
+
+
+def test_get_tracked_desktop_versions_non_string_return(mocker):
+    """_get_tracked_desktop_versions should handle non-string returns (lines 234-237)."""
+    integration = DownloadCLIIntegration()
+    mock_desktop_downloader = mocker.MagicMock()
+    mock_desktop_downloader.get_latest_release_tag.return_value = 123
+    mock_desktop_downloader.get_prerelease_tracking_file.return_value = (
+        "/tmp/desktop-prerelease.json"
```

Ruff (0.15.7)

[error] Probable insecure usage of temporary file or directory: "/tmp/desktop-prerelease.json"

(S108)